fix: Reference file storage module in terraform registry

Erin-Boehmer opened the pull request:
#### What does this implement/fix? 
byob example TF uses local path to reference the file storage module. This doesn't work for customers using the byob TF in a separate TF repo.

Instead, reference the module's path in the TF registry.

#### Comments
We might want to use a latest tag in lieu of the v1.5.3 pinned version (latest as of today).

jsbroks commented:
Thanks for the contribution!

I think the idea here is great, however I don't think this is the "terraform-ic" way, since terraform modules should only import from the `/modules` or root (`/`) directory and, this would require updating the version on each release.

I think the best apporach, would be to create a new module called `/modules/byob` which does the same as the example, using relative imports. Then we can convert to byob example to import this new module.

Im going to close this PR since we have a action plan that should satisfy your request. @venky-wandb is this something you could add to our tf?

In the meanwhile, @Erin-Boehmer have you tried importing the byob example as a module?

```terraform
module "wandb_byob" {
  source  = "wandb/wandb/aws//example/byob"
  version = "1.5.3"
  # insert the required variables here
}
```

Erin-Boehmer commented:
@jsbroks I adjusted the `main.tf` contents and added that code to our TF repo. A few changes we made:

* switch bucket-prefix from var to local 
* use existing aws provider
* change names of some resources

jsbroks commented:
#36
